lib/debugger/debugger.dart

```diff
@@ -0,0 +1,1308 @@
+        break;
+      case EventKind.kBreakpointAdded:
+        _breakpoints.value.add(event.breakpoint);
+        _breakpoints.add(_breakpoints.value);
+        break;
+      case EventKind.kBreakpointResolved:
+        _breakpoints.value.remove(event.breakpoint);
+        _breakpoints.value.add(event.breakpoint);
+        _breakpoints.add(_breakpoints.value);
+        break;
+      case EventKind.kBreakpointRemoved:
+        _breakpoints.value.remove(event.breakpoint);
+        _breakpoints.add(_breakpoints.value);
+        break;
+    }
+  }
+
+  void _clearCaches() {
+    _scriptCache.clear();
+    _lastEvent = null;
+    _reportedException = null;
+  }
+
+  void dispose() {
+    _debugSubscription?.cancel();
+  }
+
+  void _updatePaused(bool value) {
+    if (_paused.value != value) {
+      _paused.add(value);
+    }
+  }
+
+  Future<Script> getScript(ScriptRef scriptRef) async {
+    if (!_scriptCache.containsKey(scriptRef.id)) {
+      _scriptCache[scriptRef.id] =
+          await service.getObject(isolateRef.id, scriptRef.id);
+    }
+
+    return _scriptCache[scriptRef.id];
+  }
+
+  Pos calculatePosition(Script script, int tokenPos) {
+    final List<List<int>> table = script.tokenPosTable;
+    if (table == null) {
+      return null;
+    }
+
+    for (List<int> row in table) {
+      if (row == null || row.isEmpty) {
+        continue;
+      }
+      final int line = row.elementAt(0);
+      int index = 1;
+
+      while (index < row.length - 1) {
+        if (row.elementAt(index) == tokenPos) {
+          return new Pos(line, row.elementAt(index + 1));
+        }
+        index += 2;
+      }
+    }
+
+    return null;
+  }
+
+  String commonScriptPrefix;
+  LibraryRef rootLib;
+
+  void setRootLib(LibraryRef rootLib) {
+    this.rootLib = rootLib;
+
+    String scriptPrefix = rootLib.uri;
+    if (scriptPrefix.startsWith('package:')) {
+      scriptPrefix = scriptPrefix.substring(0, scriptPrefix.indexOf('/') + 1);
+    } else if (scriptPrefix.contains('/lib/')) {
+      scriptPrefix =
+          scriptPrefix.substring(0, scriptPrefix.lastIndexOf('/lib/'));
+      if (scriptPrefix.contains('/')) {
+        scriptPrefix =
+            scriptPrefix.substring(0, scriptPrefix.lastIndexOf('/') + 1);
+      }
+    } else if (scriptPrefix.contains('/bin/')) {
+      scriptPrefix =
+          scriptPrefix.substring(0, scriptPrefix.lastIndexOf('/bin/'));
+      if (scriptPrefix.contains('/')) {
+        scriptPrefix =
+            scriptPrefix.substring(0, scriptPrefix.lastIndexOf('/') + 1);
+      }
+    } else if (scriptPrefix.contains('/example/')) {
+      scriptPrefix =
+          scriptPrefix.substring(0, scriptPrefix.lastIndexOf('/example/'));
+      if (scriptPrefix.contains('/')) {
+        scriptPrefix =
+            scriptPrefix.substring(0, scriptPrefix.lastIndexOf('/') + 1);
+      }
+    } else {
+      scriptPrefix = null;
+    }
+
+    commonScriptPrefix = scriptPrefix;
+  }
+
+  String getShortScriptName(String uri) {
```

Comment:
[nit] How about `shortScriptName`

https://www.dartlang.org/guides/language/effective-dart/design#avoid-starting-a-method-name-with-get

Reply:
From the guide: "Even if the member does need to be a method because it takes arguments or otherwise isn’t a good fit for a getter, you should still avoid `get`."